juju_verify/cli.py

```diff
@@ -122,7 +156,14 @@ def parse_args() -> argparse.Namespace:
         type=str,
         help="Check all units on the machine.",
     )
-    return parser.parse_args()
+    args = parser.parse_args()
+    try:
+        parse_charm_mapping(args)
+    except ValueError:
+        parser.error(
+            "Unexpected format of --map-charm argument. For more info see --help"
+        )
+    return args
```

Comment:
We could avoid this be defining the argument `action` or `type` in `add_argument` function.